x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -21,6 +22,158 @@ import { initSavedObjects } from './saved_objects';
 import { initUsageCollectors } from './usage';
 import { OsqueryAppContext, OsqueryAppContextService } from './lib/osquery_app_context_services';
 import { ConfigType } from './config';
+import { packSavedObjectType, savedQuerySavedObjectType } from '../common/types';
+
+const registerFeatures = (features: SetupPlugins['features']) => {
+  features.registerKibanaFeature({
+    id: 'osquery',
+    name: i18n.translate('xpack.features.osqueryFeatureName', {
+      defaultMessage: 'Osquery',
+    }),
+    order: 1300,
+    category: DEFAULT_APP_CATEGORIES.management,
+    app: ['osquery', 'kibana'],
+    catalogue: ['osquery'],
+    privileges: {
+      all: {
+        app: ['osquery', 'kibana'],
+        catalogue: ['osquery'],
+        savedObject: {
+          all: ['search', 'query'],
+          read: ['index-pattern'],
+        },
+        ui: ['show', 'save'],
+      },
+      read: {
+        app: ['osquery', 'kibana'],
+        catalogue: ['osquery'],
+        savedObject: {
+          all: [],
+          read: ['index-pattern', 'search', 'query'],
+        },
+        ui: ['show'],
+      },
+    },
+    subFeatures: [
+      {
+        name: i18n.translate('xpack.features.ossFeatures.discoverShortUrlSubFeatureName', {
+          defaultMessage: 'Live queries',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'live_queries_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['crudLiveQueries', 'readLiveQueries'],
+              },
+              {
+                id: 'live_queries_read',
+                includeIn: 'read',
+                name: 'Read',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['readLiveQueries'],
+              },
```

Comment:
These two sub-feature privileges don't actually grant any permissions that are authorized on the server side.

These UI Capabilities are only intended for rendering controls differently on the client-side, but we should really be checking authorization on the server-side as well (either explicitly via `api` tags, or implicitly by granting access to saved object types).